neps/nep-0393.md

```diff
@@ -0,0 +1,596 @@
+pub struct TokenMetadata {
+    pub class: ClassId, // token class. Required. Must be non zero.
+    pub issued_at: Option<u64>, // When token was issued or minted, Unix epoch in milliseconds
+    pub expires_at: Option<u64>, // When token expires, Unix epoch in milliseconds
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+
+trait SBTRegistry {
+    /**********
+     * QUERIES
+     **********/
+
+    /// Get the information about specific token ID issued by `issuer` SBT contract.
+    fn sbt(&self, issuer: AccountId, token: TokenId) -> Option<Token>;
+
+    /// Returns total amount of tokens issued by `issuer` SBT contract, including expired
+    /// tokens. If a revoke removes a token, it must not be included in the supply.
+    fn sbt_supply(&self, issuer: AccountId) -> u64;
+
+    /// Returns total amount of tokens of given class minted by `issuer`. See `sbt_supply` for
+    /// information about revoked tokens.
+    fn sbt_supply_by_class(&self, issuer: AccountId, class: ClassId) -> u64;
+
+    /// Returns total supply of SBTs for a given owner. See `sbt_supply` for information about
+    /// revoked tokens.
+    /// If `class` is specified, returns only owner supply of the given class (either 0 or 1).
+    fn sbt_supply_by_owner(
+        &self,
+        account: AccountId,
+        issuer: AccountId,
+        class: Option<ClassId>,
+    ) -> u64;
+
+    /// Query sbt tokens issued by a given contract.
+    /// `limit` specifies the upper limit of how many tokens we want to return.
+    /// If `from_token` is not specified, then `from_token` should be assumed
+    /// to be the first valid token id. If `with_expired` is set to `true` then all the tokens are returned
+    /// including expired ones otherwise only non-expired tokens are returned.
+    fn sbt_tokens(
+        &self,
+        issuer: AccountId,
+        from_token: Option<u64>,
+        limit: Option<u32>,
+        with_expired: bool,
+    ) -> Vec<Token>;
+
+    /// Query SBT tokens by owner.
+    /// `limit` specifies the upper limit of how many tokens we want to return.
+    /// If `from_class` is not specified, then `from_class` should be assumed to be the first
+    /// valid class id. If `with_expired` is set to `true` then all the tokens are returned
+    /// including expired ones otherwise only non-expired tokens are returned.
+    /// Returns list of pairs: `(Contract address, list of token IDs)`.
+    fn sbt_tokens_by_owner(
+        &self,
+        account: AccountId,
+        issuer: Option<AccountId>,
+        from_class: Option<u64>,
+        limit: Option<u32>,
+        with_expired: bool,
+    ) -> Vec<(AccountId, Vec<OwnedToken>)>;
+
+    /// checks if an `account` was banned by the registry.
+    fn is_banned(&self, account: AccountId) -> bool;
```

Comment:
This is odd naming if you even look at it without deep context. `<contract>::is_banned` does not communicate enough context on its own. I would recommend `sbt_is_owner_banned` or something along those lines